chainer/sequential.py

```diff
@@ -229,40 +227,24 @@ def __reduce__(self):
         return super(Sequential, self).__reduce__()
 
     def __str__(self):
-        ret = ''
-        for i, layer in enumerate(self):
-            if isinstance(layer, Sequential):
-                name = layer.__class__.__name__
-                name += '\twhich has {} layers'.format(len(layer))
-            elif isinstance(layer, link.Chain):
-                name = layer.__class__.__name__
-                name += '\tThe structure behind a Chain is determined at '
-                name += 'runtime.'
-            elif isinstance(layer, link.ChainList):
-                name = layer.__class__.__name__
-                name += '\tThe structure behind a ChainList is determined at '
-                name += 'runtime.'
-            elif isinstance(layer, link.Link):
-                name = layer.__class__.__name__
-                param_info = '\t'
-                for param in sorted(layer.params(), key=lambda p: p.name):
-                    param_info += param.name
-                    if param._data[0] is not None:
-                        param_info += str(param._data[0].shape)
-                    else:
-                        param_info += '(None)'
-                    param_info += '\t'
-                name = name + param_info
-            elif isinstance(layer, function.Function):
-                name = layer.__class__.__name__
-            elif isinstance(layer, functools.partial):
-                name = repr(layer)
-            elif layer.__name__ == '<lambda>':
-                name = inspect.getsource(layer).strip()
+        reps = []
+        for index, layer in enumerate(self):
+            # Explore better representation by if-block.
+            if getattr(layer, '__name__', None) == '<lambda>':
```

Comment:
* Could you add test case for this?
* It lacks support for `functools.partial` and `function.Function`?

Reply:
> Could you add test case for this?

I added the test.